File: javascript/example_code/postgres-js/src/connection-util.js
import { generateToken } from "./token-gen.js";
import postgres from "postgres"

async function getClient(clusterEndpoint, region) {
    const action = "DbConnectSuperuser";
    let token;
    try {
        // The token expiration time is optional, and the default value 900 seconds
        token = await generateToken(clusterEndpoint, action, region);
        const sql = postgres({
            host: clusterEndpoint,
            user: "admin",
            password: token,
            database: "postgres",
            port: 5432,
            ssl: "require",
          });
        return Promise.resolve(sql)
    } catch (error) {
        return Promise.reject(error);
    }
}

export { getClient }

File: ruby/example_code/ruby-pg/lib/connection_util.rb
require 'pg'
require_relative 'token_generator'

module ConnectionUtil
    def get_connection(cluster_endpoint, region)
        action = "DbConnectSuperUser"

        credentials = Aws::SharedCredentials.new()

        begin
            token_gen = Aws::AxdbFrontend::AuthTokenGenerator.new({
                :credentials => credentials
            })
            
            # The token expiration time is optional, and the default value 900 seconds
            token = token_gen.generate_db_connect_superuser_auth_token({
                :endpoint => cluster_endpoint,
                :region => region
            })

            pg_connection = PG.connect(
                            host: cluster_endpoint,
                            user: 'admin',
                            password: token,
                            dbname: 'postgres',
                            port: 5432,
                            sslmode: 'require'
            )
            return pg_connection
        rescue => error
            raise
        end
    end
    module_function :get_connection
end
